src/sources/metadatasourcetaglib.cpp

```diff
@@ -629,22 +629,53 @@ class AiffTagSaver: public TagSaver {
  */
 class SafelyWritableFile final {
   public:
-    SafelyWritableFile(QString origFileName, bool useTemporaryFile)
-        : m_origFileName(std::move(origFileName)) {
+    SafelyWritableFile(QString origFileName, bool useTemporaryFile) {
+        // Both file names remain uninitialized until all prerequisite operations
+        // in the constructor have been completed successfully. Otherwise failure
+        // to create the temporary file will not be handled correctly!
+        // See also: https://bugs.launchpad.net/mixxx/+bug/1815305
+        DEBUG_ASSERT(m_origFileName.isNull());
         DEBUG_ASSERT(m_tempFileName.isNull());
         if (useTemporaryFile) {
-            QString tempFileName = m_origFileName + kSafelyWritableTempFileSuffix;
-            QFile origFile(m_origFileName);
-            if (origFile.copy(tempFileName)) {
-                m_tempFileName = std::move(tempFileName);
-            } else {
+            QString tempFileName = origFileName + kSafelyWritableTempFileSuffix;
+            QFile origFile(origFileName);
+            if (!origFile.copy(tempFileName)) {
                 kLogger.warning()
                         << origFile.errorString()
-                        << "- Failed to copy original into temporary file before writing:"
-                        << origFile.fileName()
+                        << "- Failed to clone original into temporary file before writing:"
+                        << origFileName
                         << "->"
                         << tempFileName;
+                // Abort constructor
+                return;
+            }
+            QFile tempFile(tempFileName);
+            DEBUG_ASSERT(tempFile.exists());
+            // Both file sizes are expected to be equal after successfully
```

Comment:
What if the changed meta-data is longer? E.g. fields are added?

Reply:
After cloning the file both files are expected to be of the same size. Nothing has been written yet.